test/integration/component/maint/test_ha_pool_maintenance.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from nose.plugins.attrib import attr
+from marvin.cloudstackTestCase import cloudstackTestCase
+from marvin.cloudstackAPI import (enableStorageMaintenance,
+                                  cancelStorageMaintenance
+                                  )
+from marvin.lib.utils import (cleanup_resources,
+                              validateList)
+from marvin.lib.base import (Account,
+                             VirtualMachine,
+                             ServiceOffering,
+                             Cluster,
+                             StoragePool,
+                             Volume)
+from marvin.lib.common import (get_zone,
+                               get_domain,
+                               get_template,
+                               list_hosts
+                               )
+from marvin.codes import PASS
+
+
+def maintenance(self, storageid):
+    """enables maintenance mode of a Storage pool"""
+
+    cmd = enableStorageMaintenance.enableStorageMaintenanceCmd()
+    cmd.id = storageid
+    return self.api_client.enableStorageMaintenance(cmd)
+
```

Comment:
hmm... I thought something like this should work: StoragePool.enableMaintenance(self.api_client, id=storageself.storageid[0][0]) since it is a classmethod
I am just trying to confirm you're using the latest base.py since couple of these enhancements went in recently into base.py...
If it still doesn't fit in, it's ok. You can continue the way you've done